### DISCUSSION TITLE
Crash Handling

### DISCUSSION THREAD

**Discussion by NikoLXQ** [getsentry/sentry-native#820] [Q&A]
Android platfrom: After a crash, why not create a new child process or child thread to handle crash info collected?Is it safe to use the async-signal-safe function in a crashed thread? eg: malloc()、calloc()、free()、snprintf()、gettimeofday()

**Comment by Swatinem** [comment]
The inproc backend does things in the signal handler mostly for simplicity reasons. And as you said, when we are already in a signal handler for whatever reason, we consider the whole process to be tainted anyway, so calling `malloc` in a different thread will not be reliable anyway, as we might as well have crashed in the allocator. The SDK is extra careful to avoid the allocator once it reached the signal handler.

The breakpad backend will spawn a ptrace-enable child process to capture a minidump, and then pass control back to the main signal handler. The crashpad backend will pass the signal to an already running detached child process for it to handle the event.

What exactly is your usecase?
